cmd/util.go

```diff
@@ -22,44 +24,53 @@ func debugPrint() {
 	}
 }
 
+func isInNamespace(meta metav1.ObjectMeta) (valid bool) {
+	return rootConfig.namespace == apiv1.NamespaceAll || rootConfig.namespace == meta.Namespace
+}
+
 func convertDeploymentToDeploymentList(deployment Deployment) (deploymentList *DeploymentList) {
-	deploymentList = &DeploymentList{
-		Items: []Deployment{deployment},
+	if isInNamespace(deployment.ObjectMeta) {
```

Comment:
I have a better suggestion for these `convertBla` should return types like `kubeAuditBla` instead of k8s types. This will help in simplifying the code. As later in the code we use convert function to construct those types. Other suggestion would be just get rid of those custom types like `kubeAuditBla` and only use native types from k8s, which would require a lot of refactoring but possible with one type like interface{}[].

Reply:
I looked at your second suggestion and it seems possible and simplifies the code. Maybe you can create a refactoring issue for that.

Reply:
I am in favor of merging this imperfect solution and address the underlying and deeper refactoring issue in another (and way way longer) PR.